src/atoms/affine/stack.jl

```diff
@@ -112,21 +112,25 @@ end
 
 # * `Value` is not owned by Convex.jl
 # * splatting creates zero-argument functions, which again are not owned by Convex.jl
+Base.hcat(args::AbstractExpr...) = HcatAtom(args...)
 
-hcat(args::AbstractExpr...) = HcatAtom(args...)
-function hcat(args::AbstractExprOrValue...)
+function Base.hcat(args::AbstractExprOrValue...)
+    if all(Base.Fix2(isa, Value), args)
+        return Base.cat(args..., dims = Val(2))
+    end
     return HcatAtom(map(arg -> convert(AbstractExpr, arg), args)...)
 end
-hcat(args::Value...) = Base.cat(args..., dims = Val(2))
 
 # TODO: implement vertical concatenation in a more efficient way
 vcat(args::AbstractExpr...) = transpose(HcatAtom(map(transpose, args)...))
 function vcat(args::AbstractExprOrValue...)
+    if all(Base.Fix2(isa, Value), args)
+        return Base.cat(args..., dims = Val(1))
```

Comment:
It I understood correctly, it seems like this code path shouldn't be reachable, as `vcat(::Value....)` is defined in Base and is a subtype (morespecific) than `vcat(::AbstractExprOrValue...)`?

Reply:
I guess it guards against that method changing in Base. But yes, it's probably not reachable in the current setup.